products/bluebell/core/src/intermediate_representation/emitter.rs

```diff
@@ -539,152 +594,158 @@ impl AstConverting for IrEmitter {
-                            right: expected_value,
-                        },
-                        source_location: source_location.clone(),
-                    });
-                    let case = self.convert_instruction_to_symbol(compare_instr);
-
-                    // Blocks for success and fail
-                    let fail_label = self
-                        .ir
-                        .symbol_table
-                        .name_generator
-                        .new_block_label("match_fail");
-
-                    let success_label = self
-                        .ir
-                        .symbol_table
-                        .name_generator
-                        .new_block_label("match_success");
-                    let mut success_block = FunctionBlock::new_from_symbol(success_label.clone());
-
-                    // Terminating current block
-                    let op = Operation::ConditionalJump {
-                        expression: case,
-                        on_success: success_label,
-                        on_failure: fail_label.clone(),
-                    };
-                    self.current_block
-                        .instructions
-                        .push_back(Box::new(Instruction {
-                            ssa_name: None,
-                            result_type: None,
-                            operation: op,
-                            source_location,
-                        }));
-                    self.current_block.terminated = true;
-
-                    // Finishing current_block and moving it onto
-                    // to the current body while preparing the success block
-                    // as current
-                    mem::swap(&mut success_block, &mut self.current_block);
-                    self.current_body.blocks.push(success_block);
-
-                    let _ = clause.node.expression.visit(self)?;
-                    let expr_instr = self.pop_instruction()?;
-                    let source_location = expr_instr.source_location.clone();
-
-                    let result_sym = self.convert_instruction_to_symbol(expr_instr);
-                    phi_results.push(result_sym);
-
-                    let exit_instruction = Box::new(Instruction {
-                        ssa_name: None,
-                        result_type: None,
-                        operation: Operation::Jump(finally_exit_label.clone()),
-                        source_location,
-                    });
-                    self.current_block.instructions.push_back(exit_instruction);
-
-                    // Pushing sucess block and creating fail block
+                unimplemented!();
+            } /* TODO: {
 
-                    let mut fail_block = FunctionBlock::new_from_symbol(fail_label.clone());
-                    mem::swap(&mut fail_block, &mut self.current_block);
-                    self.current_body.blocks.push(fail_block);
+            info!("Match statement");
```

Comment:
do you want to keep all this commented out code here? Might make sense to delete it

Reply:
Yes, I want to keep it for now as the implementation might be right. I just don't have any test case to actually test it - is not the `match` statement described in the PR but a different type of match statement.